app/lib/analyzer/task_sources.dart

```diff
@@ -4,24 +4,118 @@
 
 import 'dart:async';
 
+import 'package:async/async.dart';
+import 'package:gcloud/db.dart';
+
+import '../frontend/models.dart';
 import '../shared/task_scheduler.dart';
 
+import 'models.dart';
+
+/// Creates a task when a version uploaded in the past hour has no analysis yet.
 class DatastoreHeadTaskSource extends PollingTaskSource {
-  DatastoreHeadTaskSource() : super(new Duration(minutes: 10));
+  final DatastoreDB _db;
+  DateTime _lastTs;
+  DatastoreHeadTaskSource(this._db) : super(const Duration(minutes: 10));
 
   @override
   Stream<Task> pollTasks() {
-    // TODO: implement pollTasks
-    return new Stream.empty();
+    final DateTime now = new DateTime.now().toUtc();
+    final DateTime oneHourAgo = now.subtract(new Duration(hours: 1));
+    DateTime minCreated;
+    if (_lastTs == null) {
+      minCreated = oneHourAgo;
+    } else if (_lastTs.isBefore(oneHourAgo)) {
+      // more than an hour passed since the last poll
+      minCreated = _lastTs;
+    } else {
+      minCreated = oneHourAgo;
+    }
+    _lastTs = oneHourAgo;
+    final Query q = _db.query(PackageVersion)..filter('created >=', minCreated);
+    return q.run().asyncMap(_mapPackageVersion).where((v) => v != null);
+  }
+
+  Future<Task> _mapPackageVersion(Model m) async {
+    final PackageVersion pv = m;
+    final List<PackageVersionAnalysis> items = await _db.lookup([
+      _db.emptyKey
+          .append(PackageAnalysis, id: pv.package)
+          .append(PackageVersionAnalysis, id: pv.version)
+    ]);
+    if (items.first == null) {
+      return null;
+    } else {
+      return new Task(pv.package, pv.version);
```

Comment:
Shouldn't this be reversed? i.e. if we couldn't find an analysis, we should make a `Task` and otherwise skip it.